opendps/gen_lookup.py

```diff
@@ -123,17 +123,26 @@ def convert_font_to_c(font_fname, characters, character_strings, font_size, outp
     # Generate glyph widths
     font_source_file.write("const uint8_t font_%s_widths[%d] = {\n  " % (output_filename, len(character_strings)))
     character_max_width = 0
+    digit_max_width = 0
     count = 0
     for i in range(len(characters)):
         (width, _) = character_images[i].size
         font_source_file.write("%d" % (width))
         if width > character_max_width:
-            character_max_width = width
+            character_max_width = width # Find the maximum glyph width
+        if width > digit_max_width and characters[i].isdigit():
+            digit_max_width = width # Find the maximum digit width
         count += 1
         if count < len(characters):
             font_source_file.write(",\n  ")
     font_source_file.write("\n};\n\n")
 
+    # If no spacing parameter is set then generate one based on the max glyph width
+    if not args.font_spacing:
+        args.font_spacing = int(round(0.223 * float(character_max_width) - 0.863)) # This is fairly rough
```

Comment:
And fairly magic ;)

Reply:
I visually spaced a pile of different fonts and drew a graph from it :L

Reply:
Wow!